### ISSUE TITLE
`object-curly-spacing`: missing error for typed `ObjectPattern` without space before closing brace

### ISSUE THREAD

**Issue by kh4f** [eslint-stylistic/eslint-stylistic#1128]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The `object-curly-spacing` rule with `['error', 'always']` does not report an error for a typed `ObjectPattern` when there is no space before the closing brace of the pattern.
```ts
// src/index.ts
const foo = ({ str}: { str: string }) => null
```
- **Actual behavior**: no errors
- **Expected behavior**: `error  A space is required before '}'`

I intend to submit a PR to fix this by aligning the closing brace detection with `object-curly-newline`’s approach for `ObjectPattern` with `typeAnnotation`.

### Reproduction

https://stackblitz.com/edit/github-cnweogt9?file=src%2Findex.ts

### System Info

```shell
System:
    OS: Windows 11 10.0.26100
    CPU: (12) x64 AMD Ryzen 5 2600 Six-Core Processor
    Memory: 2.67 GB / 15.93 GB
  Binaries:
    Node: 25.2.1 - C:\nvm4w\nodejs\node.EXE
    npm: 11.6.2 - C:\nvm4w\nodejs\npm.CMD
    pnpm: 10.17.1 - C:\Users\kh4f\AppData\Local\pnpm\pnpm.EXE
  npmPackages:
    @stylistic/eslint-plugin: ^5.8.0 => 5.8.0
    eslint: ^9.39.2 => 9.39.2
```

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests
